lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -73,6 +78,522 @@ struct DiagRestore {
 static void emitDxilDiag(LLVMContext &Ctx, const char *str) {
   hlsl::dxilutil::EmitErrorOnContext(Ctx, str);
 }
+
+// Helper class which build PSV ViewID state from serialized view id state.
+struct SimpleViewIDState {
```

Comment:
This shares an awful lot with `DxilViewIdState`. Can this be made a sublclass or share the common code some other way?